crates/sp-executor/src/lib.rs

```diff
@@ -102,6 +113,62 @@ impl<Hash: Encode> BundleHeader<Hash> {
     }
 }
 
+// TODO: unify the type between the solution and threshold better.
+pub fn derive_bundle_election_solution(domain_id: DomainId, vrf_output: &[u8]) -> u64 {
+    let mut data = domain_id.to_le_bytes().to_vec();
+    data.extend_from_slice(vrf_output);
+
+    let local_domain_randomness = blake2b_256_hash(&data);
```

Comment:
There is a function that hashes a list of values, you don't need to allocate a vector like that just to hash it